golem-cli/src/factory.rs

```diff
@@ -133,6 +144,22 @@ pub trait ServiceFactory {
             clients: self.health_check_clients(auth)?,
         }))
     }
+
+    fn deploy_service(
+        &self,
+        auth: &Self::SecurityContext,
+    ) -> Result<
+        Arc<dyn DeployService<ProjectContext = Self::ProjectContext> + Send + Sync>,
```

Comment:
All the services are stateless - why would we need `Arc`?

Reply:
The service creation is quite cheap. It's just a clone for 3-4 strings. No need to keep shared service instances in `Arc` - we can just create multiple instances.

Reply:
But for example for the `--connect` it made it simpler to implement as I could clone the service to pass to the background fiber. (But that was in a previous PR)

Reply:
It is 100% up to you. And it is not realy so important any way. I just prefer to avoid Arc since it hints state sharing.